Accessibility: Removes hard-coded columns in home facet lists

ckaz opened the pull request:
* adds CSS column solution.
* based on dev branch
Reason: Lists of the same type or belonging under the same heading must be kept intact even if they are rendered as multicolumn on screen.

ckaz committed:
Accessibility: Removes hard-coded columns in home facet lists
* adds CSS column solution.
* based on dev branch

ckaz commented:
Sorry, the rebase wouldn't push so I made this new based on dev.

crhallberg commented:
So far so good. It looks like this PR is missing a change where the new variable is used. More branch/PR diff issues?

demiankatz reviewed (changes requested):
As @crhallberg notes, it looks like changes to search.scss are missing (along with the updates to compiled.css). Please let me know if I can be of any help in sorting this out!

ckaz committed:
Accessibility: Removes hard-coded columns in home facet lists

* adds missing style definitions and css files

ckaz commented:
> As @crhallberg notes, it looks like changes to search.scss are missing (along with the updates to compiled.css). Please let me know if I can be of any help in sorting this out!

Sorry, silly me. Of course! Sorry for wasting your time.
The search.scss had mostly code style changes applied. I forgot to turn this off. These should not impact functionality, but improve readability.

The key change is this:

`
.home-facet-list {
  column-count: $home-facet-list-column-count; // defines number of cols, define this in variables.scss
  list-style: none;
  padding-left: 0;
  padding-right: 1rem;

  @media (max-width: $screen-lg - 1) {
    column-count: $home-facet-list-column-count-md-sm-xs;
  }
}`

Cheers!

demiankatz commented on themes/bootstrap5/scss/components/search.scss:
```diff
@@ -45,18 +45,29 @@ $thumbnail-width-large:  160px !default;
 @media (max-width: 767px) {
   #datevispublishDatexWrapper,
-  // Bulk action checkboxes:
+    // Bulk action checkboxes:
```
For some reason, your style modifications have indented this line too far.

```suggestion
  // Bulk action checkboxes:
```

ckaz replied:
you are absolutely right

demiankatz commented on themes/bootstrap5/scss/components/search.scss:
```diff
@@ -45,18 +45,29 @@ $thumbnail-width-large:  160px !default;
   #search-cart-form .result .checkbox,
-  // Search result and favorites bulk action buttons:
-  .bulkActionButtons { display: none; }
+    // Search result and favorites bulk action buttons:
```
This one too:
```suggestion
  // Search result and favorites bulk action buttons:
```

demiankatz reviewed (changes requested):
Thanks, @ckaz. See below for some minor cosmetic issues.

A bigger issue -- this seems to be forcing two columns by default now, rather than using two columns as an overflow. With default settings on my screen, here are before and after shots:

Before:
![image](https://github.com/user-attachments/assets/803d4b0d-6032-468e-be18-39deef393f19)

After:
![image](https://github.com/user-attachments/assets/490f89a1-9ef3-469f-bd48-705cc768e1dd)

Is there a way we can prevent multiple columns from activating before they are needed? Or do you prefer it this way? (I think it looks busier and harder to read, personally).

ckaz committed:
Accessibility: Removes hard-coded columns in home facet lists

* adds column height definition so columns will break automatically beyond that height

ckaz reviewed (commented):
@demiankatz thanks for those eagle-eyed corrections!

ckaz commented:
> A bigger issue -- this seems to be forcing two columns by default now, rather than using two columns as an overflow. With default settings on my screen, here are before and after shots:

> Is there a way we can prevent multiple columns from activating before they are needed? Or do you prefer it this way? (I think it looks busier and harder to read, personally).

Indeed, we can add a custom height (introduced as a variable), after which columns break automatically.

demiankatz reviewed (changes requested):
It looks the the merge of #4224 has introduced a conflict here. Do you mind resolving it? I'll test again once that's straightened out. (This is the disadvantage to all the style changes -- they greatly increase the likelihood of conflicts while the PR is in motion).

ckaz committed:
Merge branch 'dev' of github.com:vufind-org/vufind into pullrequest_accessibility_home-facet-list_colums_dev

ckaz committed:
Accessibility: Removes hard-coded columns in home facet lists

* re-adds css files after merge

ckaz commented:
> It looks the the merge of #4224 has introduced a conflict here. Do you mind resolving it? I'll test again once that's straightened out. (This is the disadvantage to all the style changes -- they greatly increase the likelihood of conflicts while the PR is in motion).

Ok, sure. Just committed the changes. Hope that solves it.

demiankatz committed:
ExtendClass generator: Strip leading slashes from class names (#4232)

siiriylonen committed:
Open top-level singleton collection by default in hierarchy tree (#4201)

demiankatz reviewed (changes requested):
@ckaz, thanks for the latest updates. A few thoughts:

1.) If you want to keep the style improvements, it might be helpful if you could just open a separate PR with only the style fixes. I assume that doing this wouldn't be much work. We could then merge that first and subsequently reconcile changes with this PR to make the history easier to read and understand.

2.) Having a hard-coded height for the columns seems a bit at odds with having a column size setting in the FacetList content block. We still use the column height setting to decide how many values to display, but now the column splitting is based on an independent value, so the results are no longer predictable/controllable. Do you think it might be possible, rather than hard-coding the height, to instead put the column size value as a data attribute on the container element, and use Javascript to set the height dynamically based on text measurements? I realize that this is not as easy as it sounds because long text may wrap, which makes the necessary height unpredictable... but maybe there's some algorithmic way to figure out a reasonable value, given the right inputs.

Anyway, just to demonstrate why this is a concern, here are before and after shots of what happens if I set the column size to "2":

Before: 
![image](https://github.com/user-attachments/assets/b169880a-8dd8-49ec-b2ac-1eb188a548b3)

After: 
![image](https://github.com/user-attachments/assets/2d0a8534-551a-43c0-beeb-918cacafa990)

3.) If there's no reasonable way to make the new behavior equivalent to the old behavior, maybe we should consider deprecating or removing the column size option on the FacetList content block. It also occurs to me that the hard-coded special case for the callnumber-first column is ugly and would be better addressed through some kind of configuration. That's probably out of scope for this PR, but I just wanted to mention it while we're contemplating major changes here anyway.

What do you think? @crhallberg is on vacation right now, but he might also have some thoughts on this to share when he returns next week.

demiankatz committed:
Trivial: fix outdated/inaccurate comments.

demiankatz committed:
Trivial: update link/wording related to community support.

demiankatz committed:
Drop legacy 1.x compatibility from SolrMarc code. (#4234)

AnnaSmaros committed:
Update Swedish translations (#4235)

ckaz commented:
> 1.) If you want to keep the style improvements, it might be helpful if you could just open a separate PR with only the style fixes. I assume that doing this wouldn't be much work. We could then merge that first and subsequently reconcile changes with this PR to make the history easier to read and understand.

Not necessarily. We have a style linter in place that makes sure the entire team writes SCSS to the same guidelines, so code is uniformly readable. Maybe something worth considering or discussing for VuFind in general.

ckaz commented:
> 2.) Having a hard-coded height for the columns seems a bit at odds with having a column size setting in the FacetList content block. We still use the column height setting to decide how many values to display, but now the column splitting is based on an independent value, so the results are no longer predictable/controllable. Do you think it might be possible, rather than hard-coding the height, to instead put the column size value as a data attribute on the container element, and use Javascript to set the height dynamically based on text measurements? I realize that this is not as easy as it sounds because long text may wrap, which makes the necessary height unpredictable... but maybe there's some algorithmic way to figure out a reasonable value, given the right inputs.

I guess that would be the ultimate solution. I though it shouldn't be too hard to predict the FacetList entry (we have `protected $columnSize = 10;` in module/VuFind/src/VuFind/ContentBlock/FacetList.php) _and_ the height set, because  every library kind of knows their home page facets (at least roughly) enough to know what to expect. 
However, you are absolutely right. We should avoid erratic behaviour.

Since I'm almost purely an interface person, I would soon be out of my depth trying to get the column size into this. In addition, I think this might open a box of additional problem.

ckaz commented:
> 3.) If there's no reasonable way to make the new behavior equivalent to the old behavior, maybe we should consider deprecating or removing the column size option on the FacetList content block. It also occurs to me that the hard-coded special case for the callnumber-first column is ugly and would be better addressed through some kind of configuration. That's probably out of scope for this PR, but I just wanted to mention it while we're contemplating major changes here anyway.

Indeed that might be the best way forward, removing the column size option. That would leave everybody free to experiment.

demiankatz commented:
> Not necessarily. We have a style linter in place that makes sure the entire team writes SCSS to the same guidelines, so code is uniformly readable. Maybe something worth considering or discussing for VuFind in general.

I'm always in favor of consistency and good style; if there's a tool you're happy with, please feel free to open a PR if it's not too much work, and then we can discuss/adjust as needed.

demiankatz commented:
> Since I'm almost purely an interface person, I would soon be out of my depth trying to get the column size into this. In addition, I think this might open a box of additional problem.

I'm interested to hear what @crhallberg and @EreMaijala think about all this. Chris is on vacation right now, but I'll touch base with him about this when he returns.

EreMaijala commented:
I'm not sure I like a multi-column display even if it makes the page shorter. To me the result seems harder to interpret, and won't work properly with hierarchical facets. That said, I don't mind it being possible (we don't even use home page facets in Finna), but I'd rather make it opt-in for cases where it makes sense. And I suppose some of the visual dfficulty could be alleviated with e.g. background color, vertical lines separating different facets or such.

damien-git committed:
Fixed first/last_indexed, improved import perf, support for mariadb connector (#4108)

Co-authored-by: Demian Katz <demian.katz@villanova.edu>

dltj committed:
Update relator codes from Library of Congress (#4214)

Updated from https://www.loc.gov/marc/relators/relacode.html as of 28-Jan-2025

ckaz commented:
> I'm not sure I like a multi-column display even if it makes the page shorter. To me the result seems harder to interpret, and won't work properly with hierarchical facets. That said, I don't mind it being possible (we don't even use home page facets in Finna), but I'd rather make it opt-in for cases where it makes sense. And I suppose some of the visual dfficulty could be alleviated with e.g. background color, vertical lines separating different facets or such.

Hi @EreMaijala , thanks for your reply. I agree with your notion about interpreting the columns. 
However, I think we have a misunderstanding here. For accessibility for non-sighted persons who rely on screen readers, lists under the same heading must not be split _structurally_ in the html. They may be reshaped visually but must remain one `<ul>` per list or the context might get lost for people relying on screen readers. That's the entire point. I'm very happy with the columns as such (though I'm not a fan) -- we just got told off by our accessibility testing company for splitting the lists.

ckaz commented:
> > Not necessarily. We have a style linter in place that makes sure the entire team writes SCSS to the same guidelines, so code is uniformly readable. Maybe something worth considering or discussing for VuFind in general.
> 
> I'm always in favor of consistency and good style; if there's a tool you're happy with, please feel free to open a PR if it's not too much work, and then we can discuss/adjust as needed.

Cheers. It may turn into a bigger batch of things, with all the files to be revised in the end, but I'll look into that!

demiankatz commented:
> Hi @EreMaijala , thanks for your reply. I agree with your notion about interpreting the columns. However, I think we have a misunderstanding here. For accessibility for non-sighted persons who rely on screen readers, lists under the same heading must not be split _structurally_ in the html. They may be reshaped visually but must remain one `<ul>` per list or the context might get lost for people relying on screen readers. That's the entire point. I'm very happy with the columns as such (though I'm not a fan) -- we just got told off by our accessibility testing company for splitting the lists.

I think @EreMaijala's comment is just pointing out that in addition to the structural change to clarify lists for screen readers, it may be worthwhile to add a small visual cue to help separate the list for sighted users, since things kind of blend together in the current design. Something like a faint border dividing the facet field sections (like a left border on everything except the first column) might do the job. That's arguably a separate issue, but I agree that it might improve the readability.

EreMaijala commented:
@demiankatz The point I failed to make was that it requires some effort to do the multi-column layout, and I'm not sure if it's worth it. But I'm fine if people actually like it that way. :)

EreMaijala commented on themes/bootstrap5/scss/variables.scss:
```diff
@@ -3,3 +3,8 @@ $zindex-cookie-consent: ($zindex-dropdown - 10) !default;
+// Sets the number of columns in which the facets on home page should be displayed for Desktop and smaller devices
+$home-facet-list-column-height: 260px !default; // Defines the height of each column before the column breaks
+$home-facet-list-column-count: 2 !default;
+$home-facet-list-column-count-md-sm-xs: 1 !default;
```
This is sort of oddly named. In Bootstrap you'd typically have something like `$home-facet-list-column-count-md` which would have effect unless `$home-facet-list-column-count-sm` is defined and so on. Would it be too much trouble to actually support multiple breakpoints? Depending on the number of facets displayed two columns might make sense on md screen but be too much on sm.

ckaz replied:
> Would it be too much trouble to actually support multiple breakpoints? Depending on the number of facets displayed two columns might make sense on md screen but be too much on sm.

Oh, surely not. I'll look into that.

crhallberg replied:
Usually in Bootstrap they work with `min-width` media queries. So, `sm` would affect everything, unless an `md`, `lg`, etc. overrides it. You can see [Bootstrap's column documentation here](https://getbootstrap.com/docs/5.3/layout/grid/#grid-options). Based on this `-column-count` for mobile and `-column-count-md` for desktop would match the naming convention.

demiankatz commented:
> @demiankatz The point I failed to make was that it requires some effort to do the multi-column layout, and I'm not sure if it's worth it. But I'm fine if people actually like it that way. :)

The original purpose of the multi-column layout was to make it possible to display twice as many values for some facets as for others. Whether or not we need that is a separate question, but if we take away some kind of multi-column mechanism, we'll just end up with some lists being arbitrarily longer than others, which is definitely not desirable. The template is currently hard-coded to only show extra facets for call numbers, but that's something that should be made configurable if we plan to retain the functionality.

And I guess I'm also making a separate point, which is that I think separating the facet lists from each other with some visual marker might be valuable, regardless of whether or not they contain multiple columns of values (though it's definitely most important when there are multiple columns, because that makes things especially murky).

EreMaijala commented:
@demiankatz Yes, the visual separation would be useful in any case, but doesn't need to happen here for me to happy. :)

dltj committed:
Handle spaces in file names in the import scripts (#4189)

Co-authored-by: Megan Schanz <schanzme@msu.edu>

demiankatz committed:
BrowZine covers: skip over default images. (#4226)

demiankatz committed:
Pin sass package to 1.72.0. (#4242)

demiankatz committed:
Combined search: add setting to disable default filters. (#4241)

EreMaijala committed:
Make it easier to override sort order in scheduled search notify command. (#4247)

demiankatz committed:
[VUFIND-1755] Allow FOLIO token sharing between sessions. (#4244)

demiankatz committed:
Add tests to cover bottom button functionality. (#4222)

EreMaijala committed:
[10.1] Ensure that sort list always contains a selected sort option. (#4251)

EreMaijala committed:
Fix and unify record toolbar entries. (#4245)

- Remove invalid role from <li>.
- Add proper classes for icon links so that the text-decoration rule for '.action-toolbar .icon-link__label' works and provides consistency with bulk actions toolbar.
- Ensure that cart buttons use same classes as other buttons in record toolbar.
- Ensure that <a> of cart buttons are immediate children of <li> to apply correct styling.
- Remove useless styles.

demiankatz committed:
Bottom button tests: fallback with Javascript. (#4253)

demiankatz committed:
Revert "Bottom button tests: fallback with Javascript. (#4253)"

This reverts commit a038f3064f00cf7d3dae6f14d2f0bae20195f4d8.

EreMaijala committed:
Fix book bag buttons and divider (#4249)

demiankatz committed:
Trivial: delete obsolete comment.

LuomaJuha committed:
Check that href is set in HeadLink::isExcludedFromConcat (#4256)

demiankatz committed:
testBulkExport: improve reliability. (#4254)

EreMaijala committed:
Add label support to HiddenSorting options for scheduled searches. (#4252)

demiankatz committed:
Prevent malformed links in export-menu.phtml. (#4257)

ThoWagen committed:
Add option to ignore email addresses in linkify view helper (#4259)

demiankatz committed:
Trivial: correct class name in comments.

EreMaijala committed:
Tweak navbar and search tabs. (#4248)

- Aligns with Bootstrap 5's documented structure.
- Fixes accessibility issues (e.g. bad role="tablist" in search tabs that are just links from accessibility point of view).
- Reduces redundant and unnecessary css rules.
- Adds margin below search tabs (bootstrap5).
- Brings MyResearch dropdown menu button closer to the text for improved visual association.
- Adds left and right padding to main menu entries in collapsed mode for improved looks.
- Make account dropdown visible in mobile menu.

EreMaijala committed:
Trivial: Fix Mink test for account menu dropdown.

EreMaijala committed:
Fix form submission in facet sidebar to not include any hash in the URL. (#4261)

demiankatz committed:
Fix applied filter editing on non-search-results pages. (#4258)

demiankatz committed:
Make FallbackLoaderTest more robust. (#4264)

demiankatz committed:
Generalize DOI linker system to support other identifier types (#3918)

demiankatz committed:
Trivial: fix inaccurate comments.

demiankatz committed:
Remove accidentally committed file.

demiankatz committed:
Fix bug: dashes in theme names could break SCSS build. (#4265)

ThoWagen committed:
Fix record navigation when results are loaded with js (#4196)

demiankatz committed:
Prevent empty breadcrumb when record title is missing. (#4262)

EreMaijala committed:
Fix HTML validation of profile page. (#4260)

EreMaijala committed:
Clean up screen width variables and media queries. (#4263)

- Removes all screen width related Bootstrap 3 compatibility variables and classes (including utilities like .hidden-xs).
- Adds semantic classes to control visibility with styling in certain cases.
- Switches styles to use Bootstrap's media breakpoint utilities and removes hard-coded breakpoints.

maccabeelevine committed:
Fix Shibboleth examples comment (#4270)

It reads backwards to me?

demiankatz committed:
OAI server: improve exception handling (#4266)

ThoWagen committed:
Update RecordDataFormatter comment (#4268)

ThoWagen committed:
[VUFIND-1700] Use record-tab-init event instead of registerTabEvents and remove jQuery (#4250)

ThoWagen committed:
[VUFIND-1731] Aria hide sliders and replace slider library (#4238)

xmorave2 committed:
Open ID Connect authentication implementation (#1230)

EreMaijala committed:
Refactor the details dialog of feedback admin to use normal lightbox. (#4269)

EreMaijala committed:
Drop unused data from getFeedbackByFilter results (#4271)

Simplifies the query now that user_name and manager_name are no longer needed after #4269.

EreMaijala committed:
Hide additional controls from print version. (#4273)

EreMaijala committed:
Clean up legacy styles and remove bs3-compat.scss (#4267)

demiankatz committed:
Expand/modernize test for VuFind\Config\PluginFactory (#4277)

EreMaijala committed:
Move firebase/php-jwt to normal dependencies. (#4279)

Since it's now required by the OpenID Connect client, it should be installed in all environments.

EreMaijala committed:
Fix height of main content. (#4281)

EreMaijala committed:
Add a main switch for caching in CachingDownloader. (#4274)

LuomaJuha committed:
Add option to use resumptionToken in REST (#4225)

EreMaijala committed:
Fix some issues in the OpenID Connect client. (#4280)

- Improve error logging and reveal less to the user.
- Fix OIDC state handling, basic auth and JWK 'kid' handling
  - When authentication failed, the oidc_state variable was unset, causing the login link to be missing state on a subsequent request.
  - User name and secret should not be URL-encoded in basic authentication.
  - kid is optional in JWKs

ThoWagen committed:
Filter empty values in combine-alt.phtml (#4282)

rominail committed:
Multi facet selection - Record last user state (#4109)

ckaz commented:
> > > Not necessarily. We have a style linter in place that makes sure the entire team writes SCSS to the same guidelines, so code is uniformly readable. Maybe something worth considering or discussing for VuFind in general.
> > 
> > 
> > I'm always in favor of consistency and good style; if there's a tool you're happy with, please feel free to open a PR if it's not too much work, and then we can discuss/adjust as needed.
> 
> Cheers. It may turn into a bigger batch of things, with all the files to be revised in the end, but I'll look into that!

We currently uses SCSS lint as a plugin to our IDE (PhpStorm): https://github.com/idok/scss-lint-plugin  I'm sure there must be something similar for other IDEs as well.

ckaz commented:
> > @demiankatz 
>  [...]
> The original purpose of the multi-column layout was to make it possible to display twice as many values for some facets as for others. 
> [...]
> [...] I think separating the facet lists from each other with some visual marker might be valuable, regardless of whether or not they contain multiple columns of values (though it's definitely most important when there are multiple columns, because that makes things especially murky).

Sorry for the delay -- been on sick-leave ...

So do I understand the current state of discussion correctly as: 

1. It would be desirable to remove the hard-coded column breaks (`<br>` and `protected $columnSize = 10;` in module/VuFind/src/VuFind/ContentBlock/FacetList.php), which would (initially) lead to facets being output in one big list. Such lists may soon become too long for fitting visually with the other items on the page.
2. It would therefore be desirable to have an _option_ to break such long lists visually into columns. This could be done with the introduction of an (optional) SCSS variable, which every library can set to their desired value. (After all, they should know how large the lists are, locally.)
3. It would be desirable to have an option for visual separation of the facet lists from each other. This could be solved by introduction of an (optional) SCSS variable, which every library can set to their desired value. (This might be a contender for another ticket.)
4. The initially suggested variables (if maintained), such as `$home-facet-list-column-count-md-sm-xs` should be renamed/remodelled to fit better with the Bootstrap standards. (see https://github.com/vufind-org/vufind/pull/4231#discussion_r1950762293)

If so, I'll gladly revise and resubmit.

ThoWagen committed:
Fix advanced search breadcrumb (#4283)

EreMaijala committed:
Add Docker support to Vagrant. (#4284)

Adds provider-specific configuration and support for non-systemctl systems. Also adds lsof that Solr control script wants.

demiankatz commented:
> We currently uses SCSS lint as a plugin to our IDE (PhpStorm): https://github.com/idok/scss-lint-plugin I'm sure there must be something similar for other IDEs as well.

Ideally it would be best to find something we can automate through GitHub Actions to detect problems independently of IDE.

It looks like the scss-lint-plugin depends on this project: https://github.com/sds/scss-lint -- but this project has a big warning saying that we probably shouldn't use it because of deprecated dependencies. It instead recommends using this: https://stylelint.io/ -- maybe we should discuss this on a future Community Call and/or set up a JIRA ticket so we remember to investigate further.

demiankatz commented:
> Sorry for the delay -- been on sick-leave ...

Welcome back!

> So do I understand the current state of discussion correctly as:
> 
>     1. It would be desirable to remove the hard-coded column breaks (`<br>` and `protected $columnSize = 10;` in module/VuFind/src/VuFind/ContentBlock/FacetList.php), which would (initially) lead to facets being output in one big list. Such lists may soon become too long for fitting visually with the other items on the page.
> 
>     2. It would therefore be desirable to have an _option_ to break such long lists visually into columns. This could be done with the introduction of an (optional) SCSS variable, which every library can set to their desired value. (After all, they should know how large the lists are, locally.)
> 
>     3. It would be desirable to have an option for visual separation of the facet lists from each other. This could be solved by introduction of an (optional) SCSS variable, which every library can set to their desired value. (This might be a contender for another ticket.)
> 
>     4. The initially suggested variables (if maintained), such as `$home-facet-list-column-count-md-sm-xs` should be renamed/remodelled to fit better with the Bootstrap standards. (see [Accessibility: Removes hard-coded columns in home facet lists #4231 (comment)](https://github.com/vufind-org/vufind/pull/4231#discussion_r1950762293))
> 
> 
> If so, I'll gladly revise and resubmit.

This all makes sense to me. I imagine a little more tweaking will probably be needed once we see it in practice, but none of this sounds to me like a move in the wrong direction. My main concern is finding the easiest/most reliable defaults so that local customization and adjustment will be minimally necessary. I'm not up to date on the latest CSS tricks, so I don't have a specific proposal. :-)

demiankatz committed:
Trivial: fix test timing for Jenkins.

demiankatz committed:
Make SearchFacetsTest less vulnerable to timing issues.

EreMaijala committed:
Make saving of cart not fail when a record is missing. (#4288)

Previously the saving to favorites just failed with a not very helpful "Not found" error.

EreMaijala committed:
Remove Solr v3 error listener. (#4286)

EreMaijala committed:
Streamline favorite deletion from a list. (#4287)

Eliminates an extra lightbox with the success/failure message and displays it as a simple flash message on top of the main page instead.

demiankatz committed:
Remove AddThis support; service was discontinued in 2023.

demiankatz committed:
Test timing fix, attempt number three.

EreMaijala committed:
Allow NoILS to validate in installer without further configuration. (#4285)

ThoWagen committed:
Refactor "copy to clipboard" button and use it for citations (#4289)

demiankatz committed:
Trivial: normalize tabs to spaces in templates.

ThoWagen committed:
[VUFIND-1614] Modernize PubDateVisAjax recommendation module (#4275)

EreMaijala committed:
Bump version for 10.1.2 release. (#4294)

EreMaijala committed:
Run composer install in the build directory. (#4293)

ThoWagen committed:
Fix search-filter-dropdown button icon (#4296)

EreMaijala committed:
Exclude macOS special files and extended attributes from packaged tar files. (#4295)

EreMaijala committed:
Add Solr account handling and clean up output of deb package (#4292)

demiankatz committed:
Improve/modernize RecordTab/VersionsTest (#4297)

demiankatz committed:
[VUFIND-747] Create Breadcrumbs view helper. (#4236)

demiankatz committed:
Fix bug: delete language command incompatible with single-quotes. (#4298)

demiankatz committed:
Configuration upgrade fixes. (#4299)

demiankatz committed:
Bump version number for release 10.1.3.

ckaz committed:
Accessibility: Removes hard-coded columns in home facet lists

* revises home page facet colum display by introducing use of css columns for callnumber-first facets
* makes use of single column layout for all other home page facets
* introduces visual division
* all customizable via SCSS

ckaz commented:
So, finally, after a lot of back-and-forth, here's my updated proposal:

1. Until a better solution is found, _we maintain the option that the number of entries is set in `module/VuFind/src/VuFind/ContentBlock/FacetList.php`_ (currently 10 entries,  `$columnSize = 10` , after which a link is inserted, which is misleadingly called "More options ..." but leads to "Advanced Search"). 
3. Since all I could glean from the code is that _only  `callnumber-first` are to be displayed as two columns_, I've simplified my suggestion to _one column for all home facets_ except `callnumber-first` , which gets two columns. The column number is set via SCSS variables (along with a bunch of other settings which may be useful). Using variables allows every library to define their own custom settings. 
The two-column option can be overridden (again using SCSS variables) for devices from 0-420px, 420-767px, 767px-991px and above. Of course if can be defaulted to 3 or more columns, too.
In addition, _other exceptions for displaying home facets in multiple columns could be added_ by changing the  setting in `ContenBlock/FacetList.phtml` (currently: `$cssColumnClass = $field == 'callnumber-first' ? 'two-columns' : 'one-column'; `)
4. The multi-column setting uses CSS column breaks to satisfy W3C accessibility requirements.
5. For safety and variability, there are additional settings defining, for example, height (defaulting to auto), which can be combined with `clip` which defines what will happen if entries overflow the defined height.
6. There is a _customizable visual separation_ now, using a vertical line (with color, width + line type) + padding. All of this is defined via SCSS variables, so it can be set to none, or to custom colors, widths etc

Pls. let me know what you think and whether it is feasible to continue considering this.
Cheers, Claas

ckaz commented:
Sorry, I rebased to get all the in-between changes and I may have initially pushed to the wrong repo. So this looks a little unruly now. I can gladly redo it from scratch ...

ckaz committed:
Accessibility: Removes hard-coded columns in home facet lists

* re-adds bootstrap5 changes

demiankatz commented:
@ckaz, I think you need to merge the dev branch into your branch to reduce GitHub's confusion. If it's easier to close this and start again, though, that's fine too! It will be easier to continue reviewing and discussing your latest proposal once we can more easily view the relevant diffs. :-)

ckaz commented:
Will redo from scratch

demiankatz commented:
Superseded by #4302.
